applications/MeshingApplication/custom_utilities/parmmg/pmmg_utilities.cpp

```diff
@@ -0,0 +1,1413 @@
+
+        /* Elements */
+        std::size_t num_tetra = 0, num_prisms = 0;
+        for(IndexType i = 0; i < r_elements_array.size(); ++i) {
+            auto it_elem = it_elem_begin + i;
+
+            if ((it_elem->GetGeometry()).GetGeometryType() == GeometryData::KratosGeometryType::Kratos_Tetrahedra3D4) { // Tetrahedron
+                for (auto& r_node : it_elem->GetGeometry()) {
+                    total_nodes_to_remesh.insert(r_node.Id());
+                }
+                num_tetra += 1;
+            } else {
+                KRATOS_ERROR << "ParMmg currently only supports tetrahedras on elements. Your geometry type was: " << (it_elem->GetGeometry()).GetGeometryType() <<  std::endl;
+            }
+        }
+
+        pmmg_mesh_info.NumberOfTetrahedra = num_tetra;
+        pmmg_mesh_info.NumberOfPrism = num_prisms;
+
+        KRATOS_INFO_IF("ParMmgUtilities", ((num_tetra + num_prisms) < r_elements_array.size()) && GetEchoLevel() > 0) <<
+        "Number of Elements: " << r_elements_array.size() << " Number of Tetrahedron: " << num_tetra << " Number of Prisms: " << num_prisms << std::endl;
+    }
+
+    pmmg_mesh_info.NumberOfNodes = total_nodes_to_remesh.size();
+    SetMeshSize(pmmg_mesh_info);
+
+    // We reorder the ids to avoid conflicts with the rest (using as reference the OLD_ENTITY)
+    IndexType counter_to_remesh = 0;
+    #pragma omp parallel for reduction(+: counter_to_remesh)
+    for(int i = 0; i < static_cast<int>(r_nodes_array.size()); ++i) {
+        auto it_node = it_node_begin + i;
+
+        const bool old_entity = it_node->IsDefined(OLD_ENTITY) ? it_node->Is(OLD_ENTITY) : false;
+        if (!old_entity) {
+            ++counter_to_remesh;
+        }
+    }
+
+    mGlobalToLocalNodePreMap = std::unordered_map<int, int>();
+    mGlobalToLocalElemPreMap = std::unordered_map<int, int>();
+    mGlobalToLocalCondPreMap = std::unordered_map<int, int>();
+
+    IndexType counter_remesh = 0;
+    for(int i = 0; i < static_cast<int>(r_nodes_array.size()); ++i) {
+        auto it_node = it_node_begin + i;
+        mGlobalToLocalNodePreMap[it_node->Id()] = ++counter_remesh;
+    }
+    counter_remesh = 0;
+    for(int i = 0; i < static_cast<int>(r_conditions_array.size()); ++i) {
+        auto it_cond = it_cond_begin + i;
+        mGlobalToLocalCondPreMap[it_cond->Id()] = ++counter_remesh;
+    }
+    counter_remesh = 0;
+    for(int i = 0; i < static_cast<int>(r_elements_array.size()); ++i) {
+        auto it_elem = it_elem_begin + i;
+        mGlobalToLocalElemPreMap[it_elem->Id()] = ++counter_remesh;
+    }
+
+    // Now we compute the colors
+    rColors.clear();
+    ColorsMapType nodes_colors, cond_colors, elem_colors;
+    AssignUniqueModelPartCollectionTagUtility model_part_collections(rModelPart);
+    model_part_collections.ComputeTags(nodes_colors, cond_colors, elem_colors, rColors);
+
+    /* Nodes */
+    #pragma omp parallel for firstprivate(nodes_colors)
```

Comment:
Do parallelism work here (Meaning, is really faster?) I don't think they can  implement a SetXXX without a lock inside.

Reply:
I see, you are probaly right. This was kept as in mmg_utilities. Do you recommend removing it?